numpy/core/tests/test_conversion_utils.py

```diff
@@ -0,0 +1,124 @@
+"""
+Tests for numpy/core/src/multiarray/conversion_utils.c
+"""
+import pytest
+
+import numpy as np
+import numpy.core._multiarray_tests as mt
+
+
+class StringConverterTestCase:
+    allow_bytes = True
+    case_insensitive = True
+    exact_match = False
```

Comment:
this is a bit weird, why do we support partial matches, seems like it can cause confusion with user code when C can be used for order as well as mode.

Reply:
I agree it's weird, but my goal of this PR was to test the behavior that's already present. I wouldnt be opposed to someone else championing for deprecations once this is in.

Reply:
Indeed, Every time I saw it, I thought it should be deprecated, but it also never felt super pressing. Which doesn't mean it shouldn't be done, just that I didn't particularly feel like actually doing it when stumbling over it :).